frontend_tests/casper_tests/12-subscription_announcement.js

```diff
@@ -0,0 +1,30 @@
+var common = require('../casper_lib/common.js').common;
+
+common.start_and_log_in();
+
+casper.then(function () {
+    casper.test.info('Narrowing to stream notifications');
+    casper.click('a[href^="#narrow/stream/notifications"]');
+});
+
+
+casper.then(function () {
+    casper.test.assertSelectorHasText('title', 'notifications - Zulip Dev - Zulip', 'Narrowed to stream notifications');
+    casper.test.assertTextExists('Bot created', "Stream Announcement exists");
+    this.clickLabel('Subscribe to announcement', 'button');
+});
+
+casper.waitForText('Unsubscribe from announcement', function () {
+    casper.test.assertTextExists('Unsubscribe from announcement', "Subscribed to Announcement");
+    this.clickLabel('Unsubscribe from announcement', 'button');
+});
+
+casper.waitForText('Subscribe to announcement', function () {
+    casper.test.assertTextExists('Subscribe to announcement', "Unsubscribed from Announcement");
+});
```

Comment:
Ahh, so you're testing a different thing than I expected.  What this should be testing is the following sequence:

(1) go to streams page, create a new stream, selecting the "announce stream" option.

(2) Narrow to the announcement stream ("zulip" in this case), and verify that a notification bot announcement was sent there

(3) subscribe/unsubscribe using the button in the announcement from (2).